apps/apps.c

```diff
@@ -2566,3 +2566,55 @@ int opt_printf_stderr(const char *fmt, ...)
     va_end(ap);
     return ret;
 }
+
+OSSL_PARAM *app_params_new_from_opts(STACK_OF(OPENSSL_STRING) *opts,
+                                     const OSSL_PARAM *paramdefs)
+{
+    OSSL_PARAM *params = NULL;
+    size_t sz = (size_t)sk_OPENSSL_STRING_num(opts);
+    size_t params_n;
+    int ok = 1;
+
+    if (opts == NULL)
+        return NULL;
+
+    params = OPENSSL_zalloc(sizeof(OSSL_PARAM) * (sz + 1));
+    if (params == NULL)
+        return NULL;
+
+    for (params_n = 0; params_n < sz; params_n++) {
+        char *opt = sk_OPENSSL_STRING_value(opts, (int)params_n);
+        char *stmp, *vtmp = NULL;
+
+        if ((stmp = OPENSSL_strdup(opt)) == NULL
+            || (vtmp = strchr(stmp, ':')) == NULL
+            || (*vtmp++ = 0) /* Always zero */
```

Comment:
DONT DO THIS.  Take the time to add the few lines and make the code clear and obvious.

Reply:
Uhmmm, it's fairly obvious C, really. Of course, reading this requires some rudimentary C knowledge, but considering how many other languages gave copied the exact same behavior, I find the claim of unclarity surprising....

Reply:
You suggested the comma operator in another spot. Do you mean this?

``` C
            || (*vtmp++ = 0, 
                OSSL_PARAM_allocate_from_text(&params[params_n], paramdefs,
                                              stmp, vtmp, strlen(vtmp))) {
```